v0/destinations/pipedrive/transform.js

```diff
@@ -0,0 +1,570 @@
+    const value = {
+      amount: payload.amount,
+      currency: payload.currency
+    };
+    set(payload, "value", value);
+  }
+  delete payload.amount;
+  delete payload.currency;
+
+  const response = defaultRequestConfig();
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.endpoint = endpoint;
+  response.params = {
+    api_token: Config.apiToken
+  };
+  response.headers = {
+    "Content-Type": "application/json",
+    Accept: "application/json"
+  };
+
+  return response;
+};
+
+async function responseBuilderSimple(message, category, destination) {
+  let builderResponse;
+
+  switch (category.type) {
+    case EventType.IDENTIFY:
+      builderResponse = await identifyResponseBuilder(
+        message,
+        category,
+        destination
+      );
+      break;
+
+    case EventType.GROUP:
+      builderResponse = await groupResponseBuilder(
+        message,
+        category,
+        destination
+      );
+      break;
+
+    case EventType.ALIAS:
+      builderResponse = await aliasResponseBuilder(
+        message,
+        category,
+        destination
+      );
+      break;
+
+    case EventType.TRACK:
+      builderResponse = await trackResponseBuilder(
+        message,
+        category,
+        destination
+      );
+      break;
+
+    default:
+      throw new CustomError("invalid event type", 400);
+  }
+
+  return builderResponse;
+}
+
+async function process(event) {
+  const { message, destination } = event;
+  let category;
+
+  if (!message.type) throw new CustomError("message type is invalid", 400);
+
+  const messageType = message.type.toLowerCase().trim();
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      category = CONFIG_CATEGORIES.IDENTIFY;
+      break;
+    case EventType.ALIAS:
+      // dummy since alias does have a mapping json
+      category = { type: "alias" };
+      break;
+    case EventType.GROUP:
+      category = CONFIG_CATEGORIES.GROUP;
+      break;
+    case EventType.TRACK:
+      category = CONFIG_CATEGORIES.TRACK;
+      break;
+    default:
+      throw new CustomError("invalid message type", 400);
+  }
+  const res = await responseBuilderSimple(message, category, destination);
+  return res;
+}
+
+const processRouterDest = async inputs => {
+  if(!Array.isArray(inputs) || inputs.length === 0) {
+    const respEvents = getErrorRespEvents(null, 400, "Invalid events array");
+    return [respEvents];
+  }
+
+  const respList = await Promise.all(
+    inputs.map(async input => {
+      try {
+        return getSuccessRespEvents(
+          await process(input),
+          [input.metadata],
+          input.destination
+        );
+      } catch (error) {
+        return getErrorRespEvents(
+          [input.metadata],
+          error.response
+            ? error.response.status
+            : error.code
+            ? error.code
+            : 500,
```

Comment:
should we return 500/400?

Reply:
yeah I think we can make it 400 also.  Since, most 500 related cases are already covered in the code, it seems.